xarray/core/dataarray.py

```diff
@@ -386,14 +386,15 @@ def _replace(
         variable: Variable = None,
         coords=None,
         name: Union[Hashable, None, Default] = _default,
+        indexes=None,
     ) -> "DataArray":
         if variable is None:
             variable = self.variable
         if coords is None:
             coords = self._coords
         if name is _default:
             name = self.name
-        return type(self)(variable, coords, name=name, fastpath=True)
+        return type(self)(variable, coords, name=name, fastpath=True, indexes=indexes)
```

Comment:
Do we need to do this in `.copy` (which I think calls this method)?

Reply:
I can do that in a followup.

Reply:
Great, yes only if needed, 👍  re small PRs in